tests/app/questionnaire/test_rules.py

```diff
@@ -426,3 +426,30 @@ def test_should_minus_one_from_maximum_repeats(self):
         number_of_repeats = evaluate_repeat(repeat, answer_store)
 
         self.assertEqual(number_of_repeats, 24)
+
+    def test_evaluate_when_rules_condition_is_not_met(self):
+        # Given
+        when = {
+            'id': 'next-question',
+            'when': [
+                {
+                    'id': 'my_answers',
+                    'condition': 'not set',
+                    'value': '2'
+                },
+                {
+                    'id': 'my_answers',
+                    'condition': 'not set',
+                    'value': '2'
+                }
+            ]
+        }
+
+        answer_store = AnswerStore()
+        answer_store.add(Answer(answer_id='my_answers', value='1'))
+
+        # When
+        with self.assertRaises(Exception) as err:
+            evaluate_when_rules(when['when'], None, answer_store, 0)
```

Comment:
To test this exception, you should have multiple answers with the same answer_id in the answer store